### PR TITLE
Update ControllerConfig deprecation notice to point DeploymentRuntimeConfig

### LINKED ISSUE #4844: Update `ControllerConfig` deprecation warning to recommend using `DeploymentRuntimeConfig`
### What problem are you facing?

We emit a deprecation warning when folks use a `ControllerConfig`. This was added before runtime config was implemented so right now it just links to the design.

### How could Crossplane help solve your problem?

We've now implemented `DeploymentRuntimeConfig` support. We should tell folks to use it.

### PR DESCRIPTION
**Pull request by turkenh**
### Description of your changes

It changes from:

```bash
❯ kubectl create -f controller-config.yaml
Warning: ControllerConfig.pkg.crossplane.io/v1alpha1 is deprecated. It is safe to use, but no new features will be added. Future replacement design: https://github.com/crossplane/crossplane/blob/master/design/one-pager-package-runtime-config.md
controllerconfig.pkg.crossplane.io/cc-debug created
```

To:

```bash
❯ kubectl create -f controller-config.yaml
Warning: ControllerConfig.pkg.crossplane.io/v1alpha1 is deprecated. Use DeploymentRuntimeConfig from pkg.crossplane.io/v1beta1 instead.
controllerconfig.pkg.crossplane.io/cc-debug2 created
```

Fixes #4844

I have: 

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] ~Added or updated unit tests.~
- [ ] ~Added or updated e2e tests.~
- [ ] ~Linked a PR or a [docs tracking issue] to [document this change].~
- [ ] ~Added `backport release-x.y` labels to auto-backport this PR.~

Need help with this checklist? See the [cheat sheet].

[contribution process]: https://github.com/crossplane/crossplane/tree/master/contributing
[docs tracking issue]: https://github.com/crossplane/docs/issues/new
[document this change]: https://docs.crossplane.io/contribute/contribute
[cheat sheet]: https://github.com/crossplane/crossplane/tree/master/contributing#checklist-cheat-sheet

### CODE REVIEW

**Review by jeanduplessis — APPROVED** [review]

File: cluster/local/kind.sh [hunk 1]
```diff
@@ -72,7 +72,7 @@ case "${1:-}" in
   helm-upgrade)
     echo "copying image for helm"
     helm_tag="$(cat _output/version)"
-    copy_image_to_cluster ${BUILD_IMAGE} "${PROJECT_NAME}/${PROJECT_NAME}:${helm_tag}" "${KIND_NAME}"
+    copy_image_to_cluster ${BUILD_IMAGE} "xpkg.upbound.io/${PROJECT_NAME}/${PROJECT_NAME}:${helm_tag}" "${KIND_NAME}"
```
Comment (phisco) [inline cluster/local/kind.sh:75]: Looks like you also brought in the change from the other pr

**Comment by ytsarev** [comment]
Will we give customers some migration window to switch from `ControllerConfig` to `DeploymentRuntimeConfig`?
If yes, for how many releases? Can we reflect this info in the deprecation message?

File: cluster/local/kind.sh (hunk 1 above)
Comment (turkenh) [reply cluster/local/kind.sh]: Removed those from here.

File: apis/pkg/v1alpha1/config.go [hunk 2]
```diff
@@ -176,11 +176,12 @@ type PodObjectMeta struct {
...
 // ControllerConfig is the CRD type for a packaged controller configuration.
-// Deprecated: This API is scheduled to be removed in a future release. See
-// https://github.com/crossplane/crossplane/issues/3601 for more information.
+// Deprecated: This API is replaced by DeploymentRuntimeConfig, and is scheduled
+// to be removed in a future release. See the design doc for more details:
+// https://github.com/crossplane/crossplane/blob/11bbe13ea3604928cc4e24e8d0d18f3f5f7e847c/design/one-pager-package-runtime-config.md
 // +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
 // +kubebuilder:resource:scope=Cluster
-// +kubebuilder:deprecatedversion:warning="ControllerConfig.pkg.crossplane.io/v1alpha1 is deprecated. It is safe to use, but no new features will be added. Future replacement design: https://github.com/crossplane/crossplane/blob/master/design/one-pager-package-runtime-config.md"
+// +kubebuilder:deprecatedversion:warning="ControllerConfig.pkg.crossplane.io/v1alpha1 is deprecated. Use DeploymentRuntimeConfig from pkg.crossplane.io/v1beta1 instead."
```
Comment (pedjak) [inline apis/pkg/v1alpha1/config.go:184]: do we have a plan to remove it at some point, or we just want to discourage people to use it? If the former, could we decide in what Crossplane version this is gone and announce in this message, similar to what k8s does?

Also, if we want to remove it, then we should definitely put that note in the release announcements and the doc.

Comment (turkenh) [reply apis/pkg/v1alpha1/config.go:184]: I think I've responded to this [above](https://github.com/crossplane/crossplane/pull/4846#issuecomment-1770261698) 🤔 Or am I missing something?

**Pull request merged**